code/modules/mob/observer/ghost/ghost.dm

```diff
@@ -30,6 +30,9 @@ var/global/list/image/ghost_sightless_images = list() //this is a list of images
 	incorporeal_move = 1
 
 /mob/observer/ghost/New(mob/body)
+	if (istype(body, /mob/observer/ghost))
+		return//A ghost can't become a ghost.
+
```

Comment:
That doesn't seem like good idea. `..()` will be passed and other setup like `ghost_multitool = new(src)`
Ghost shouldn't become ghost in normal condition, but that's better to log such situations and handle them in any other proc. Current solution only make thing worse.

Reply:
Its not the most graceful method, but it does stop the stuff farther down in the proc.
It was mainly intended as a safety to prevent runtime errors and things getting horribly broken

Reply:
> prevent runtime errors and things getting horribly broken

`new /mob/observer/ghost (ghost)` is not "horribly broken things" it's common thing.

Reply:
doing